main/tasks/dns_task.cpp

```diff
@@ -0,0 +1,89 @@
+#include "dns_task.h"
+#include <lwip/sockets.h>
+#include <lwip/netdb.h>
+#include <esp_log.h>
+#include <time.h>
+#include <string>
+#include <map>
+#include <mutex>
+
+static const char* TAG = "dns_task";
+
+#ifndef DNS_CACHE_TTL_SECONDS
+#define DNS_CACHE_TTL_SECONDS 3600
+#endif
+
+struct CachedEntry {
+    ip_addr_t ip;
+    time_t timestamp;
+};
+
+static std::map<std::string, CachedEntry> dns_cache;
```

Comment:
Hmm I'm not so keen on using such complet dynamic memory things from the STL. Imho we don't need a STL map for just cashing a single string :sweat_smile: 

In overall the functions look a lot more complicated than it need to be